db/migrate/20180711212702_add_militarystatus_to_user.rb

```diff
@@ -0,0 +1,5 @@
+class AddMilitarystatusToUser < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :military_status, :string
```

Comment:
As this is a `string`, is the plan to accept any `string`, or a specific list of strings for consistency?  Meaning a specified set of four statuses, for example.  I **highly** recommend the latter.  

This way, we can enforce data integrity, hang logic on the responses, do data analysis, filtering and searching, add rules for certain statuses, etc., etc.

If the answer is yes, then we'll need to do a few extra additions in this PR before shipping it.

Reply:
I'd just need to add the check in `/app/models/user.rb`, is that correct?

Reply:
@hollomancer - To establish the specific set of validated statuses, you'll want to:

1. Create a constant for each status [here](https://github.com/OperationCode/operationcode_backend/blob/master/app/models/user.rb#L22).  
2. Create a constant to represent an array of all of the statuses.  You'll need to include one for `nil`, to account for our current users, and future users, that will not have a status.  For example:

```ruby
CURRENT = 'current'
VETERAN = 'veteran'
SPOUSE = 'spouse'
MILITARY_STATUSES = [CURRENT, VETERAN, SPOUSE, nil]
```

3. In `models/user.rb`, Add an `inclusion` validation for the new `:military_status` attribute, for the `MILITARY_STATUSES`.  [Here is how to do that.](http://guides.rubyonrails.org/active_record_validations.html#inclusion) 
4. In [`admin/user.rb`](https://github.com/OperationCode/operationcode_backend/blob/master/app/admin/user.rb#L93), add the set of selections to ActiveAdmin dashboard **form** portion.  For example:

```ruby
f.input :military_status, as: :select, collection: User::MILITARY_STATUSES.map { |status| [status, status].compact }.reject(&:empty?), include_blank: true
```

5. Add test coverage to `user_test.rb` testing the new inclusion validation
6. In [apiary.apib](https://github.com/hollomancer/operationcode_backend/blob/c0d852e8481d229d520652c7051567ff5af4ed7c/apiary.apib#L1810), in the `military_status` definition, change that to something like:

```
Users military status, either: current, veteran, spouse, or blank
```